packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -250,7 +250,7 @@ export default class {
   async connectXenServer(id) {
     const server = (await this._getXenServer(id)).properties
 
-    const xapi = new Xapi({
+    const xapi = (this._xapis[server.id] = new Xapi({
```

Comment:
Remove this entry if the connection fails.

Also fix this PR title, make it clear what it is fixing.